File: project-euler/euler02.py
#!/usr/bin env python2.7

# calculate the sum of even-valued terms of the Fibonacci sequence
# for all such terms less than 4 million

max_term = 4 * 10**6

prev_term = 1
cur_term = 2

fib_sum = 0

while cur_term < max_term:
  if cur_term % 2 == 0:
    fib_sum += cur_term
    print 'added: ', cur_term

  cur_term += prev_term
  prev_term = cur_term - prev_term
  print 'current: ', cur_term
  print 'prevous: ', prev_term

print fib_sum

File: project-euler/euler02.java
/**
 * Solution to Project Euler #2:
 * By considering the terms in the Fibonacci sequence whose values
 * do not exceed four million, find the sum of the even-valued terms.
 * 
 * 3-14-12
 * @author Ryan McCann
 */

import java.math.BigInteger;

public class euler02
{
	final static int Four_Million = 4000000;
	public static void main(String[] args)
	{
		BigInteger a = BigInteger.ONE;
		BigInteger b = BigInteger.ONE;
		long total = 0;
		
		while(a.max(b).intValue() < Four_Million)
		{
			a = a.add(b);
			if(a.intValue()%2 == 0){
				total += a.intValue();
			}
			
			b = a.add(b);
			if(b.intValue()%2 == 0){
				total += b.intValue();
			}
		}
		
		System.out.println(total);
	}
}
